File: jbake-core/src/main/java/org/jbake/template/model/PublishedCustomExtractor.java
package org.jbake.template.model;

import org.jbake.app.ContentStore;
import org.jbake.app.DocumentList;
import org.jbake.model.DocumentModel;
import org.jbake.model.TemplateModel;

public class PublishedCustomExtractor implements ModelExtractor<DocumentList<DocumentModel>> {

    String customDocumentType;

    public PublishedCustomExtractor(String customDocumentType) {
        this.customDocumentType = customDocumentType;
    }

    @Override
    public DocumentList<DocumentModel> get(ContentStore db, TemplateModel model, String key) {
        return db.getPublishedContent(customDocumentType);
    }

}

File: jbake-core/src/main/java/org/jbake/template/model/PublishedCustomExtractor.kt
package org.jbake.template.model

import org.jbake.app.DocumentList
import org.jbake.template.TypedModelExtractor

class PublishedCustomExtractor(private val customDocumentType: String) : TypedModelExtractor<DocumentList<*>> {
    override fun extract(context: RenderContext, key: String): DocumentList<*> {
        return context.db.getPublishedContent(customDocumentType)
    }
}
